src/decorator.js

```diff
@@ -0,0 +1,16 @@
+import shouldPureComponentUpdate from './function';
+import { Component } from 'react';
+
+export default function PureRender(Component) {
+  const componentName = Component.name || 'component';
+
+  if (!(Component.prototype instanceof Component)) {
+    throw new Error(`${componentName} is not a React component`);
```

Comment:
Why not? Don't all React components have to inherit from `Component`? It wouldn't make sense to apply this to an ES5-style React component.